docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,300 @@
+
+Both plaintext and ciphertext messages are authenticated by the sender.
+
+==== Removing inactive operators
+
+If Alice fails to receive the expected message from Bob
+within the specified timeout for the phase,
+she broadcasts this information to the rest of the group.
+If the majority of participants agree that Bob failed to send the message,
+Bob is added to the list of inactive members.
+
+If the majority does not agree that Bob failed to send the message,
+they will re-broadcast Bob's message to ensure that all members receive it.
+
+Whenever an operator is deemed to be inactive,
+the remaining participants start another key generation attempt
+with all inactive members removed.
+
+==== Dealing with misbehaviour
+
+The protocol in `tss-lib` is not attributable.
+When a participant is unable to proceed
+because they have received an invalid message from another participant,
+they will broadcast a special message declaring a _disqualification abort_
+and naming one or more participants who sent invalid messages.
+
+When a _disqualification abort message_ is broadcast by any member,
+group selection is aborted and no further attempts are made by the group.
+Instead, the group signs and submits an on-chain transaction
+announcing that key generation was aborted due to disqualification,
+and identifying the participants in the aborted attempt,
+the members who declared a disqualification abort
+and the members who were accused of sending invalid messages.
+All participants will also store their logs of the aborted attempt
+to facilitate later investigation.
+These logs must contain all messages sent and received by the participant,
+private keys the participant had established with all other participants,
+and all secret values the participant generated for the key generation attempt.
+
+Governance will then resolve the matter by investigating the stored logs
+and taking any action deemed appropriate as permitted by the contract.
+Governance may penalise any operators
+named in a disqualification abort transaction
+as either the accusing or accused parties.
+
+The `tss-lib` protocol is intended to be eventually replaced
+with a fully attributable key generation protocol.
+When the key generation protocol is replaced,
+the powers of governance to adjudicate misbehaviour off-chain
+and penalise operators deemed to have misbehaved
+are to be removed in favour of a fully objective resolution method.
+
+==== Successful key generation
+
+When an attempt at key generation finishes successfully,
+a transaction announcing this result is submitted on-chain.
+The transaction contains the public key from the successful DKG attempt,
+the list of participants in that successful attempt,
+and signatures from a majority of participants.
+
+The on-chain contract verifies
+that all listed participants are among the originally selected members,
+and that all provided signatures are valid signatures from different members.
```

Comment:
Do we expect signatures from all members not marked as inactive (previously validating this number >=  C)? Or are C signatures enough?

Reply:
T signatures is enough